### PR TITLE
Upgrade plan equipment assignments

### PR DESCRIPTION
**Pull request by Jebzou**

### CODE REVIEW

File: ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/EquipmentAssignment/EquipmentAssignmentItemModel.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using ElectronicObserverTypes;
+
+namespace ElectronicObserver.Window.Tools.EquipmentUpgradePlanner.EquipmentAssignment;
+
+public class EquipmentAssignmentItemModel
+{
+	public int Id { get; set; }
+
+	public EquipmentUpgradePlanItemModel Plan { get; set; } = null!;
```
Comment (myangelkamikaze) [inline ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/EquipmentAssignment/EquipmentAssignmentItemModel.cs:9]: Make it a required property so you don't need the `= null!;` part.
`public required EquipmentUpgradePlanItemModel Plan { get; set; }`

File: ElectronicObserver/Window/Dialog/EquipmentPicker/EquipmentPickerViewModel.cs [hunk 2]
```diff
@@ -17,8 +17,8 @@ public abstract partial class EquipmentPickerViewModel : WindowViewModelBase
 	protected abstract List<IEquipmentData> AllEquipments { get; }
 
 	public IEquipmentData? SelectedEquipment { get; set; }
-
-	protected EquipmentPickerViewModel()
+
```
[the remark this answers is not in the document]
Comment (Jebzou) [reply ElectronicObserver/Window/Dialog/EquipmentPicker/EquipmentPickerViewModel.cs]: done

File: ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/EquipmentUpgradePlanItemViewModel.cs [hunk 3]
```diff
@@ -118,6 +122,7 @@ public EquipmentUpgradePlanItemViewModel(EquipmentUpgradePlanItemModel plan)
 		EquipmentPicker = Ioc.Default.GetService<EquipmentPickerService>()!;
 		TimeChangeService = Ioc.Default.GetService<TimeChangeService>()!;
```
Comment (myangelkamikaze) [inline ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/EquipmentUpgradePlanItemViewModel.cs:123]: Ah, can you change those to GetRequiredService and remove the ! at the end?

File: ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/UpgradeTree/UpgradeTreeViewModel.cs [hunk 4]
```diff
@@ -7,7 +8,7 @@ namespace ElectronicObserver.Window.Tools.EquipmentUpgradePlanner.UpgradeTree;
 public class UpgradeTreeViewModel : WindowViewModelBase
 {
 	public ObservableCollection<UpgradeTreeUpgradePlanViewModel> Items { get; } = new();
-
+
```
Comment (myangelkamikaze) [inline ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/UpgradeTree/UpgradeTreeViewModel.cs:11]: Useless empty space.

File: ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/EquipmentUpgradePlanItemViewModel.cs (hunk 3 above)
Comment (Jebzou) [reply ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/EquipmentUpgradePlanItemViewModel.cs]: done

File: ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/UpgradeTree/UpgradeTreeViewModel.cs (hunk 4 above)
Comment (Jebzou) [reply ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/UpgradeTree/UpgradeTreeViewModel.cs]: Ctrl + K, D'd

File: ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/UpgradeTree/UpgradeTreeUpgradePlanViewModel.cs [hunk 5]
```diff
@@ -118,107 +114,25 @@ public void CleanupUnusedPlan()
 			child.CleanupUnusedPlan();
 		}
 	}
-
-	private void InitializeFromConversion(EquipmentUpgradePlanItemViewModel plan)
+
```
Comment (myangelkamikaze) [inline ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/UpgradeTree/UpgradeTreeUpgradePlanViewModel.cs:117]: Useless space.

Comment (Jebzou) [reply ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/UpgradeTree/UpgradeTreeUpgradePlanViewModel.cs]: fixed

**Review by myangelkamikaze — APPROVED** [review]

**Pull request merged**